awx/api/serializers.py

```diff
@@ -2201,6 +2201,89 @@ def create(self, validated_data):
         return return_data
 
 
+class BulkHostDeleteSerializer(serializers.Serializer):
+    hosts = serializers.ListField(
+        allow_empty=False,
+        max_length=250,
+        write_only=True,
+        help_text=_('List of hosts ids to be deleted, e.g. [105, 130, 131, 200]'),
+    )
+
+    class Meta:
+        model = Host
+        fields = 'hosts'
+        read_only_fields = ()
+
+    def validate(self, attrs):
+        request = self.context.get('request', None)
```

Comment:
```suggestion
        request = self.context.get('request', None)
        max_hosts = settings.BULK_HOST_MAX_DELETE
```
then uses `max_hosts` later on in function. Each access of `settings` is expensive/can fire queries to DB

Reply:
done